loaders/android/bootstrap.c.in

```diff
@@ -77,19 +89,38 @@ jint JNI_OnLoad(JavaVM* vm, void* reserved){
 JNIEnv* GetJNIEnv(){
   int error=0;
   JNIEnv* env = NULL;
-  if (s_vm) error=(*s_vm)->AttachCurrentThread(s_vm, &env, NULL);
-  if (!error&&(*env)->ExceptionCheck(env)) return NULL;
+  /* static `env` does NOT work!  Once in a while we should ponder if
+     it still does not work or why.
+
+  if(env) {
+    if((*env)->ExceptionCheck(env)) (*env)->ExceptionClear(env);
+    return env;
+  }
+  */
+  if(s_vm) error=(*s_vm)->AttachCurrentThread(s_vm, &env, NULL);
+  //if(!error&&(*env)->ExceptionCheck(env)) return NULL;
+  if(!error) error = JNI_forward_exception_to_gambit(env);
   return (error?NULL:env);
 }
 
+int JNI_forward_exception_to_gambit(JNIEnv*env) {
+  // TBD: actually forward, not only clear!
+  if((*env)->ExceptionCheck(env)) {
+    (*env)->ExceptionClear(env);
```

Comment:
So a call into `log_c` with content as described in https://stackoverflow.com/questions/10408972/how-to-obtain-a-description-of-a-java-exception-in-c-when-using-jni ?